prov/sm2/src/sm2_fifo.h

```diff
@@ -241,4 +241,30 @@ static inline void sm2_fifo_write_back(struct sm2_ep *ep,
 	sm2_fifo_write(ep, xfer_entry->hdr.sender_gid, xfer_entry);
 }
 
+static inline const char *sm2_print_ptr(struct sm2_ep *ep, void *ptr)
+{
+	static char outstr[80];
+	struct sm2_av *av;
+	struct sm2_mmap *map;
+	av = container_of(ep->util_ep.av, struct sm2_av, util_av);
+	map = &av->mmap;
+	struct sm2_coord_file_header *header = (void *) map->base;
+	if (map->base > (char *) ptr)
+		sprintf(outstr, "%p (not a mapped pointer, address before map)",
+			ptr);
+	else if (map->base + map->size < (char *) ptr)
+		sprintf(outstr, "%p (not a mapped pointer, address after map)",
+			ptr);
+	else {
+		int64_t relptr = sm2_absptr_to_relptr(ptr, map);
+		int jregion = (relptr - header->ep_regions_offset) /
+			      header->ep_region_size;
+		int64_t region_off =
+			(int64_t) ptr - (int64_t) sm2_peer_region(ep, jregion);
+		sprintf(outstr, "%p (%ld, %ld bytes into region %d)", ptr,
+			relptr, region_off, jregion);
+	}
+	return outstr;
```

Comment:
Returned local variable isn't accessible after the function exits. See what the fi_tostr function does to get around it.

Reply:
The variable is static, so it should be allocated in the data segment.  I suppose we are uselessly pre-allocating it even though we don't expect it to be used (this is primarily for debug support convenience).  I'll make it a NULL pointer and allocate it if needed.

Reply:
on second thought that might make valgrind mad.  Let's just leave it as is.